godot-core/src/obj/base.rs

```diff
@@ -35,6 +35,10 @@ pub struct Base<T: GodotClass> {
 }
 
 impl<T: GodotClass> Base<T> {
+    pub(crate) unsafe fn from_base(base: &Base<T>) -> Base<T> {
+        Base::from_obj(Gd::from_obj_sys_weak(base.obj_sys()))
+    }
```

Comment:
`unsafe` is a keyword to push invariant checking from the compiler onto the developer (in this case, the caller of the function).

If you declare an `unsafe` function, you need to document in a `# Safety` section under what circumstances it is sound to use. In particular:
* how do you make sure this doesn't outlive the Godot object (or is not accessed after its destruction)
* how do you avoid double destruction (the two `Base` instances)

Let me know if you need some support here 🙂

Reply:
i made this function tbf so it's a bit on me that it doesn't have a safety doc.

something like
```rs
/// 
```

Reply:
I would additionally mention that `Base<T>` holds a weak reference, i.e. its destruction does not cause destruction of the object.

Reply:
Added the comment in, lemme know if you want it reworded.

Reply:
Just punctuation nitpicking 😉 

> The returned `Base` is a weak pointer so holding it will not keep the object alive and it must not be accessed after the object is destroyed.

=> 

> The returned `Base` is a weak pointer, so holding it will not keep the object alive. It must not be accessed after the object is destroyed.